internal/olm/operator/registry/index_image.go

```diff
@@ -91,6 +92,29 @@ func (c IndexImageCatalogCreator) UpdateCatalog(ctx context.Context, cs *v1alpha
 		}
 		prevRegistryPodName = annotations[registryPodNameAnnotation]
 	}
+
+	existingItems, err := getExistingBundleItems(cs.GetAnnotations())
+	if err != nil {
+		return fmt.Errorf("error getting existing bundles from CatalogSource %s annotations: %v", cs.GetName(), err)
+	}
+	annotationsNotFound := len(existingItems) == 0
+
+	if annotationsNotFound {
+		if cs.Spec.Image == "" {
+			// if no annotations exist and image reference is empty, error out
+			return errors.New("cannot upgrade: no catalog image reference exists in catalog source spec or annotations")
+		}
+		// if no annotations exist and image reference exists, set it to index image
+		c.IndexImage = cs.Spec.Image
+	}
+
+	updateFields := func(*v1alpha1.CatalogSource) {
```

Comment:
This looks like a bug in the existing implementation
```suggestion
	updateFields := func(cs *v1alpha1.CatalogSource) {
```